lib/sendf.c

```diff
@@ -254,6 +255,43 @@ ssize_t Curl_send_plain(struct connectdata *conn, int num,
                         const void *mem, size_t len, CURLcode *code)
 {
   curl_socket_t sockfd = conn->sock[num];
+#ifdef _WIN32
+  /* WinSock will destroy unread received data if send() is
+     called after detection of disconnect.
+     To avoid lossage of received data, recv() must be
+     performed before every send() if any incoming data is
+     available. */
+  if((conn->handler->protocol&PROTO_FAMILY_HTTP) != 0 &&
+      conn->recv[num] == Curl_recv_plain) {
+    const int readymask = Curl_socket_check(sockfd, CURL_SOCKET_BAD,
+                                            CURL_SOCKET_BAD, 0);
+    if(readymask != -1 && (readymask & CURL_CSELECT_IN) != 0) {
+      struct SessionHandle * const data = conn->data;
+      struct UrlState * const st = &data->state;
+      if(!st->postponed_buffer) {
+        /* Use buffer default size for intermediate buffer */
+        st->postponed_buffer_size = data->set.buffer_size ?
+                                    data->set.buffer_size : BUFSIZE;
```

Comment:
We could add more defines or use dynamically growing buffer, but I'd like to simplify things.
A real usecase for this code - is catching small server error responses. If server sends 1GB file and closes connection before libcurl completed sending request - it will not be handled by this code.
Could use some constant here and simplify code by removing `postponed_buffer_size`. But I think it's better to keep as is for readability (just need better variable names).